src/main/kotlin/ch/bfh/habits/services/HabitService.kt

```
@@ -10,62 +10,28 @@ import org.springframework.stereotype.Service
 
-        val newHabit = HabitEntityBuilder.createHabitEntityFromDTO(habitDTO, userId)
+    @Transactional fun b(u: Long) = h.findAllByUserId(u)
 
-        if (habitDTO.groupId != null) {
-            val group = groupService.getGroup(habitDTO.groupId!!, userId)
-            newHabit.group = group
-        }
-
-        return habitDAO.save(newHabit)
+    @Transactional fun c(d: HabitDTO, u: Long): Habit {
+        if ((d.frequencyValue != null && d.frequency == null) || (d.frequencyValue == null && d.frequency != null)) throw BadRequestException("Frequency and frequencyValue must be set together")
```

detekt (1.23.7)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)